lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -3709,7 +3709,7 @@ LogicalResult FIRRTLLowering::visitDecl(InstanceOp oldInstance) {
 
   if (newInstance.getInnerSymAttr())
     if (auto forceName = circuitState.instanceForceNames.lookup(
-            {cast<hw::HWModuleOp>(newInstance->getParentOp()).getNameAttr(),
+            {newInstance->getParentOfType<hw::HWModuleOp>().getNameAttr(),
```

Comment:
Nice cleanup. Always feel free to land this kind of change directly (especially since it is nicely factored into a separate commit!).